azerothcore/azerothcore-wotlk
# fix(Core/BG): do not start new BGs if there are BGs with free slots

## CHANGES PROPOSED:

- Prevent new Battlegrounds to start if there are already other Battlegrounds that have free slots  
- Happened mostly when a lot of players queued at the same time (for example when a BG just finished)


## ISSUES ADDRESSED:

- Closes https://github.com/azerothcore/mod-cfbg/issues/11


## TESTS PERFORMED:
- Build, run, etc... (OK)
- Tried to reproduce the issue before the fix (OK)
- Tried to reproduce the issue after the fix (OK)
- Also tried to make a BG start when the current BG running is full (OK)
- Also tried BG with bracket <80 (OK)


## HOW TO TEST THE CHANGES:

For testing purposes, you can:

```
UPDATE battleground_template SET MinPlayersPerTeam = 1
UPDATE battleground_template SET MaxPlayersPerTeam = 2
```

This will make a BG start with 1 player per team, and will consider a BG as "full" when there are 2 players per team inside the BG.

Then in order to test it:

1) Try to start a BG
2) Try to add players to it until it's full
3) Try to make another BG start

Also you can try to reproduce the original issue:

- Try to make multiple players join BG at the same time, only 1 BG should start until it's full

## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
